core/src/main/java/org/jruby/RubyDir.java

```diff
@@ -420,6 +451,22 @@ public static IRubyObject foreach19(ThreadContext context, IRubyObject recv, IRu
         return foreachCommon(context, recv, context.runtime, pathString, encoding, block);
     }
 
+    private static IRubyObject eachChildCommon(ThreadContext context, IRubyObject recv, Ruby runtime, RubyString _path, RubyEncoding encoding, Block block) {
+        if (block.isGiven()) {
+            RubyClass dirClass = runtime.getDir();
+            RubyDir dir = (RubyDir) dirClass.newInstance(context, new IRubyObject[]{_path}, block);
+
+            dir.each_child(context, block);
```

Comment:
Does this path (block passed in) in MRI really ignore encoding param?

Reply:
Good observation. I'm not really sure about MRI on this, but ``Dir::foreach`` JRuby method implementation is currently doing something similar if I'm not mistaken: https://github.com/jruby/jruby/blob/ruby-2.5/core/src/main/java/org/jruby/RubyDir.java#L481

I'll get back to this later with my findings.